src/main/java/space/obminyashka/items_exchange/dao/AdvertisementRepository.java

```diff
@@ -19,6 +21,9 @@
 
     Iterable<Advertisement> findFirst10ByTopicIgnoreCaseContaining(String topic);
 
+    @Query("SELECT a from Advertisement a where (a.id = :id) and (a.user.username = :username)")
+    Optional<Advertisement> findAdvertisementByIdAndUsername(long id, String username);
+
```

Comment:
You could use Spring Data expression without @Query: 
`Optional<Advertisement> findAdvertisementByIdAndUserUsername(long id, String username)`